mesos-client/src/test/scala/mesosphere/mesos/client/MesosClientIntegrationTest.scala

```diff
@@ -0,0 +1,145 @@
+package mesosphere.mesos.client
+
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, Source}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.IntegrationTest
+import mesosphere.marathon.integration.setup.MesosClusterTest
+import mesosphere.mesos.conf.MesosConf
+import org.apache.mesos.v1.mesos.{Filters, FrameworkID, FrameworkInfo}
+import org.apache.mesos.v1.scheduler.scheduler.Event
+import org.scalatest.Inside
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.Future
+
+@IntegrationTest
+class MesosClientIntegrationTest extends AkkaUnitTest
+  with MesosClusterTest
+  with Eventually
+  with Inside
+  with StrictLogging {
+
+  "Mesos client should successfully subscribe to mesos without framework Id" in withFixture() { f =>
+    When("a framework subscribes")
+
+    val res: Future[Seq[Event]] = f.client.mesosSource.take(1).runWith(Sink.seq)
+
+    Then("a framework successfully subscribes without a framework Id")
+    inside(res.futureValue) {
+      case events =>
+        events.head.`type` shouldBe Some(Event.Type.SUBSCRIBED)
+    }
+
+    And("connection context should be initialized")
+    inside(f.client.contextPromise.future.futureValue) {
+      case ConnectionContext(host, port, mesosStreamId, frameworkId) =>
+        host shouldBe f.conf.mesosMasterHost
+        port shouldBe f.conf.mesosMasterPort
+        mesosStreamId shouldBe defined
+        frameworkId shouldBe defined
+    }
+  }
+
+  "Mesos client should successfully subscribe to mesos with framework Id" in {
+    val frameworkID = FrameworkID(UUID.randomUUID().toString)
+
+    When("a framework subscribes with a framework Id")
+    withFixture(Some(frameworkID)) { f =>
```

Comment:
Out of curiosity, why is the `withFixture` here and not at the beginning of the test?

Reply:
it gets the `frameworkID` passed as parameter